src/Components/ServiceScene/JSONPanel/LogsJsonComponent.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useCallback, useRef } from 'react';
+
+import { css } from '@emotion/css';
+
+import { FieldType, GrafanaTheme2 } from '@grafana/data';
+import { AdHocFilterWithLabels, SceneComponentProps, sceneGraph } from '@grafana/scenes';
+import { Alert, Badge, PanelChrome, useStyles2 } from '@grafana/ui';
+
+import { isLogLineField } from '../../../services/fields';
+import { getLogsHighlightStyles } from '../../../services/highlight';
+import {
+  setJsonHighlightVisibility,
+  setJsonLabelsVisibility,
+  setJsonMetadataVisibility,
+} from '../../../services/store';
+import { getFieldsVariable, getJsonFieldsVariable, getLineFiltersVariable } from '../../../services/variableGetters';
+import { LogsPanelHeaderActions } from '../../Table/LogsHeaderActions';
+import { NoMatchingLabelsScene } from '../Breakdowns/NoMatchingLabelsScene';
+import LabelRenderer from '../JSONPanel/LabelRenderer';
+import ValueRenderer from '../JSONPanel/ValueRenderer';
+import { LogListControls } from '../LogListControls';
+import { LogsJsonScene } from '../LogsJsonScene';
+import { LogsListScene } from '../LogsListScene';
+import { getLogsPanelFrame } from '../ServiceScene';
+import GetItemString from './GetItemString';
+import { JSONTree } from '@gtk-grafana/react-json-tree';
+
+export default function LogsJsonComponent({ model }: SceneComponentProps<LogsJsonScene>) {
+  const {
+    data,
+    emptyScene,
+    hasJsonFields,
+    jsonFiltersSupported,
+    menu,
+    showHighlight,
+    showLabels,
+    showMetadata,
+    sortOrder,
+  } = model.useState();
+  const $data = sceneGraph.getData(model);
+  // Rerender on data change
+  $data.useState();
+  const logsListScene = sceneGraph.getAncestor(model, LogsListScene);
+  const { visualizationType } = logsListScene.useState();
+  const styles = useStyles2((t) => getStyles(t, showHighlight));
```

Comment:
```suggestion
  const styles = useStyles2(getStyles, showHighlight);
```

Reply:
Oh this is nice, I've got to remember this one!